tests/postgres_tests/test_hstore.py

```diff
@@ -208,6 +208,13 @@ def test_invalid_json(self):
         self.assertEqual(cm.exception.messages[0], 'Could not load JSON data.')
         self.assertEqual(cm.exception.code, 'invalid_json')
 
+    def test_badly_formatted_json(self):
+        field = forms.HStoreField()
+        with self.assertRaises(exceptions.ValidationError) as cm:
+            field.clean('["a", "b", 1]')
+            self.assertEqual(cm.exception.messages[0], 'Input should resemble a dictionary.')
```

Comment:
Use `asserRaisesMessage` for the message part, keep `cm` around for the `code` assertion.